### PR TITLE
feat: Nhost typescript plugin

### PR DESCRIPTION
**Pull request by plmercereau**
## Description

Typescript plugin for the [Nhost GraphQL client to come](https://github.com/nhost/nhost/pull/1587).
As some team members from The Guild already know, Nhost is about to release a client that generates GraphQL operations from a serializable object. Both this object and the GraphQL result are strongly typed from the schema generated by this plugin.

What the plugin does:
- Generate types with `graphql-codegem/typescript`, following a strict configuration. (the plugin does not require `@graphql-codegen/typescript` to be installed by the user but is use as a dependency).
- Minify the introspection object with `'@urql/introspection`
- Export the introspection object as well as all the types so they can be accessed through a default `import schema from './generated-schema'`

[Related documentation PR](https://github.com/dotansimha/graphql-code-generator/pull/8975)

## Type of change

- [x] New plugin

## Sandbox:

[Preview of how the client will work](https://stackblitz.com/edit/nhost-graphql-client?file=play%2Findex.test.ts)

## How Has This Been Tested?

Follows the Guild guidelines

**Test Environment**:

- OS: Mac OSX
- NodeJS: 16

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/plugins/typescript/nhost/package.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "name": "@graphql-codegen/typescript-nhost",
+  "version": "0.0.1",
```
Comment (ardatan) [inline packages/plugins/typescript/nhost/package.json:3]:
```suggestion
  "version": "0.0.0",
```
We need to let changesets to make the release. So let's create a `patch` changeset using `yarn changeset` on the root

Comment (plmercereau) [reply packages/plugins/typescript/nhost/package.json]: done

File: packages/plugins/typescript/nhost/tests/nhost.spec.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    schema {
+      query: Query
+      mutation: Mutation
+      subscription: Subscription
+    }
+  `);
+
+  describe('Nhost plugin', () => {
+    it('should generate the nhost-compatibel schema', async () => {
```
Comment (ardatan) [inline packages/plugins/typescript/nhost/tests/nhost.spec.ts:88]:
```suggestion
    it('should generate the nhost-compatible schema', async () => {
```

File: packages/plugins/typescript/nhost/tests/nhost.spec.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import { buildSchema } from 'graphql';
+import { plugin } from '../src/index.js';
+
+describe('TypeScript Operations Plugin', () => {
```
Comment (ardatan) [inline packages/plugins/typescript/nhost/tests/nhost.spec.ts:4]:
```suggestion
describe('TypeScript Nhost Plugin', () => {
```

Comment (ardatan) [reply packages/plugins/typescript/nhost/tests/nhost.spec.ts:4]: Or this is extra in general?

Comment (plmercereau) [reply packages/plugins/typescript/nhost/tests/nhost.spec.ts]: no, your suggestion makes sense

Comment (ardatan) [reply packages/plugins/typescript/nhost/tests/nhost.spec.ts]: Then this is extra;
https://github.com/dotansimha/graphql-code-generator-community/pull/269/files#diff-980d5a880ad6793f2ff5f0f02eeb168103ea4c72f7d082c427c3e3e288b467b7R87

Comment (plmercereau) [reply packages/plugins/typescript/nhost/tests/nhost.spec.ts]: corrected

**Review by saihaj — APPROVED** [review]

**Pull request merged**